### ISSUE TITLE
Subscript and superscript not exposed with non-default styles

### ISSUE THREAD

**Issue by aardrian** [nvaccess/nvda#14124]
1. NVDA 2022.2 with Firefox 105 does not expose superscript text with CSS style `vertical-align: text-top`.
2. NVDA 2022.2 with Firefox 105 does not expose subscript text with CSS style `vertical-align: baseline`.

## Steps to reproduce:

1. Enable sub/superscript announcement in NVDA (Preferences → Settings… → Document Formatting → Font: Superscript and subscripts).
2. Ooen Firefox and browse to https://cdpn.io/pen/debug/xxjOrEX.
3. Compare announcements of subscript and superscript text in each of the four sections.

## Actual behavior:

- Superscript text with CSS style `vertical-align: text-top` is not announced as superscript.
- Subscript text with CSS style `vertical-align: baseline` is not announced as subscript.

## Expected behavior:

Regardless of CSS vertical alignment styles, I expect any text in `<sup>` and `<sub>` to be announced as such.

## Additional Info:

The browser default styling for `<sub>` is `vertical-align: sub` and for `<sup>` is `vertical-align: super`. With browser default styles the element is exposed.

The sub- and super-script text is exposed when paired with Chrome, but I see no distinction in the accessibility tree nor inspector in either browser on why the nodes should be announced differently. _So I am not sure if this is an NVDA issue or a Firefox issue_.

### System info:

- Windows 11
- NVDA 2022.2
- Firefox 105
- Chrome 105

### Test case:

- Cruft-free Codepen debug mode: https://cdpn.io/pen/debug/xxjOrEX
- Using Codepen editor: https://codepen.io/aardrian/pen/xxjOrEX

### Transcript of announcements:

This is a transcript from the NVDA speech viewer.

Superscript at Ascender Line (`vertical-align: super`)
- 42  superscript  nd  baseline
- superscript  2  baseline  /  subscript  3  baseline
- superscript  a  baseline  /  subscript  c  baseline
- this  superscript  A  baseline
- E=mc  superscript  2  baseline
- μ  superscript  +  baseline
- superscript  238  baseline  U,
- superscript  c  baseline  /  subscript  o  baseline 
- H  subscript  2  baseline  O,
- H  subscript  2  baseline  SO  subscript  4

Superscript above Ascender Line (`vertical-align: text-top`)
- 42nd
- 2/  subscript  3  baseline 
- a/  subscript  c  baseline
- thisA
- E=mc2
- μ+
- 238U
- c/  subscript  o  baseline
- H  subscript  2  baseline  O
- H  subscript  2  baseline  SO  subscript  4

Subscript at Baseline (`vertical-align: baseline`)
- 42  superscript  nd  baseline
- superscript  2  baseline  /3
- superscript  a  baseline  /c
- this  superscript  A  baseline
- E=mc  superscript  2  baseline
- μ  superscript  +  baseline
- superscript  238  baseline  U
- superscript  c  baseline  /o
- H2O
- H2SO4

Subscript below Baseline (`vertical-align: sub`)
- 42  superscript  nd  baseline
- superscript  2  baseline  /  subscript  3  baseline
- superscript  a  baseline  /  subscript  c  baseline
- this  superscript  A  baseline
- E=mc  superscript  2  baseline
- μ  superscript  +  baseline
- superscript  238  baseline  U
- superscript  c  baseline  /  subscript  o  baseline
- H  subscript  2  baseline  O
- H  subscript  2  baseline  SO  subscript  4

**Comment by feerrenrut** [comment]
This seems to be an issue in FireFox, the IAccessible attribute `text-position: super` is not exposed.

**Comment by jcsteh** [comment]
https://bugzilla.mozilla.org/show_bug.cgi?id=1787976

**Comment by aardrian** [comment]
Thank you both! Since James opened an issue with Firefox, I will close this.
